pydis_site/apps/api/models/bot/aoc_link.py

```diff
@@ -0,0 +1,20 @@
+from django.db import models
+
+from pydis_site.apps.api.models.bot.user import User
+from pydis_site.apps.api.models.mixins import ModelReprMixin
+
+
+class AocAccountLink(ModelReprMixin, models.Model):
+    """An AoC account link for a Discord User."""
```

Comment:
Can only linked users be blocked from being a completionist? Is that separate from this table because we need to be able to keep a user blocked if they unlink their account?

If that is the case, it may be better to merge the tables, and instead of deleting the record when the user unlinks, just set `aoc_username` to null/empty. This way, the block will remain for unlinked users. What are everyone's thoughts on this alternative?

Reply:
No, I don't believe so, this way Admins can block users even if they haven't linked their AoC account.
Otherwise, they would be limited to wait until they link their account, and then block them.

Reply:
Regardless of the general approach proposed, I'd still part with using two tables, to keep things consistent, and easier to handle on the bot side, as they're two distinct commands.